R/functions_netCDF.R

```diff
@@ -1,98 +1,278 @@
+#'                         Y_WGS84 = rep(c(47.90625, 47.96875, 48.03125, 48.09375,
+#'                                    48.15625), 2))
+#'
+#' crs <- '+init=epsg:4326'
+#'
+#' annual_time_bounds <- c(43737, 44102, 44102, 44467, 44467, 44832, 44832, 45197,
+#'                         45197, 45563, 45563, 45928, 45928, 46293, 46293, 46658,
+#'                         46658, 47024, 47024, 47389) # beginning and end of year
+#'
+#' outFileName <- 'dummynetcdf.nc'
+#'
+#' # define attributes ---------------------------------------------------------
+#'
+#' # time attribute
+#' time_attributes <- list(
+#'    name = 'time',
+#'    units = 'days since 1900-01-01',
+#'    calendar = 'standard',
+#'    unlim = TRUE,
+#'    vals = c(43554, 43920, 44285, 44650, 45015, 45381, 45746, 46111, 46476, 46842)# mid point of year
+#'    )
+#'
+#' # variable attributes
+#' var_attributes <- list(
+#'    name = 'JulyTemp',
+#'    long_name = 'Annual Mean July Temperature',
+#'    units = 'Celsius',
+#'    description = 'example data!'
+#'    )
+#'
+#' # global attributes
+#' global_attributes <- list(
+#'    title = "",
+#'    institution = 'Southwest Biological Science Center, U.S. Geological Survey',
+#'    description = 'how this data was made',
+#'    source = paste(
+#'      "SOILWAT2 (v4.2.0);",
+#'      "rSOILWAT2 (v2.3.2);",
+#'      "rSFSW2 (v3.1.2)"
+#'    ),
+#'    source_id = "SOILWAT2",
+#'    realm = "land",
+#'    parent_mip_era = "CMIP5",
+#'    parent_experiment_id = "RCP45",
+#'    parent_source = "CanESM2",
+#'    parent_variant_label = "r1i1p1f1",
+#'    product = "model-output",
+#'    projection = 'Geographic',
+#'    grid = 'WGS84',
+#'    grid_label = "gn",
+#'    nominal_resolution = "10 km", # \code{\link{calculate_nominal_resolution}}
+#'    further_info_url = "https://github.com/DrylandEcology/",
+#'    contact = "you@email.com"
+#'    )
+#'
+#' # run function -------------------------------------------------------------
+#' create_empty_netCDF_file(
+#'      data = data,
+#'      has_T_timeAxis = TRUE,
+#'      has_Z_verticalAxis = FALSE,
+#'      time_bounds = annual_time_bounds,
+#'      vert_bounds = NULL,
+#'      var_attributes = var_attributes,
+#'      time_attributes  = time_attributes,
+#'      vertical_attributes = NULL,
+#'      global_attributes = global_attributes,
+#'      isGridded = TRUE,
+#'      locations = locations,
+#'      crs = crs,
+#'      file = outFileName,
+#'      force_v4 = TRUE,
+#'      overwrite = TRUE
+#'    )
+#'
+#' @seealso \code{\link{populate_netcdf_from_array}}
+#' @seealso \url{http://cfconventions.org/cf-conventions/cf-conventions.html} # for defining attributes
+#' @seealso Demonstration of netCDF functionality in rSW2analysis vignette
 #'
 #' @export
-create_netCDF_from_raster_with_variables <- function(x, time_bounds,
-  var_attributes, global_attributes, file, force_v4 = TRUE,
-  overwrite = FALSE) {
 
+create_empty_netCDF_file <- function(data, has_T_timeAxis = FALSE,
```

Comment:
Could we add a verbose = FALSE argument and wrap the messages into `if (verbose) message(...)`?

Reply:
fixed